Savanna/Model/Animal.cs

```diff
@@ -0,0 +1,170 @@
+        public static void MakeNextRandomMove(Animal animal, List<AnimalNewCoordinates> newAnimalsCoordinates)
+        {
+            Random random = new();
+            AnimalNewCoordinates nextCoordinatesToMove = newAnimalsCoordinates[random.Next(0, newAnimalsCoordinates.Count)];
+            animal.RowCoordinate = nextCoordinatesToMove.NewRowCoordinate;
+            animal.ColumnCoordinate = nextCoordinatesToMove.NewColumnCoordinate;
+        }
+
+        /// <summary>
+        /// Calculates possible free cells on board to move animal. 
+        /// </summary>
+        /// <param name="animals">List of animals.</param>
+        /// <param name="animal">Current animal.</param>
+        /// <returns>List of free cells for possible move for current animal.</returns>
+        protected static List<AnimalNewCoordinates> FreeCellsToMove(List<Animal> animals, Animal animal)
+        {
+            List<AnimalNewCoordinates> newAnimalsCoordinates = new();
+
+            /// <summary>
+            /// Contains new animal coordinates for possible next move.
+            /// </summary>
+            AnimalNewCoordinates newCoordinates;
+
+            for (int newColumnCoordinate = animal.ColumnCoordinate - 1; newColumnCoordinate <= animal.ColumnCoordinate + 1; newColumnCoordinate++)
+            {
+                for (int newRowCoordinate = animal.RowCoordinate - 1; newRowCoordinate <= animal.RowCoordinate + 1; newRowCoordinate++)
+                {
+                    if (!IsCellReserved(newColumnCoordinate, newRowCoordinate, animals) && !Game.IsCellOnBoard(newColumnCoordinate, newRowCoordinate))
+                    {
+                        newCoordinates = new AnimalNewCoordinates
+                        {
+                            NewColumnCoordinate = newColumnCoordinate,
+                            NewRowCoordinate = newRowCoordinate
+                        };
+
+                        newAnimalsCoordinates.Add(newCoordinates);
+                    }
+                }
+            }
+
+            newCoordinates = new AnimalNewCoordinates
+            {
+                NewColumnCoordinate = animal.ColumnCoordinate,
+                NewRowCoordinate = animal.RowCoordinate
+            };
+
+            newAnimalsCoordinates.Add(newCoordinates);
+            return newAnimalsCoordinates;
+        }
+
+        /// <summary>
+        /// Checks if cell is reserved.
+        /// </summary>
+        /// <param name="newRowCoordinate">New row position.</param>
+        /// <param name="newColumnCoordinate">New column position.</param>
+        /// <param name="animals">List of animals.</param>
+        /// <returns>Is cell reserved.</returns>
+        private static bool IsCellReserved(int newRowCoordinate, int newColumnCoordinate, List<Animal> animals)
+        {
+            return animals.Where(animal => animal.RowCoordinate.Equals(newRowCoordinate) && animal.ColumnCoordinate.Equals(newColumnCoordinate)).Any();
```

Comment:
This can be simplified like
```c#
return animals.Any(animal => animal.RowCoordinate.Equals(newRowCoordinate) && animal.ColumnCoordinate.Equals(newColumnCoordinate));
```
You can resolve this comment yourself when fixed.

Reply:
Ok, Thank you!